### PR TITLE
lib: Add a private helper to abort txns, use in sysroot cleanup

### PR DESCRIPTION
**Pull request by cgwalters**
Steal some code from flatpak for this, which allows porting a few more things to
new style. I started on a public API version of this but was trying to roll some
other things into it and it snowballed.

Let's do this version since it's easy for now.

Note that while here I changed things so that `generate_deployment_refs()` now
uses a single txn rather than 4.

### CODE REVIEW

File: src/libostree/ostree-sysroot-cleanup.c [hunk 1]
```diff
@@ -353,34 +354,18 @@ cleanup_ref_prefix (OstreeRepo         *repo,
                     GCancellable       *cancellable,
                     GError            **error)
 {
```
Comment (jlebon) [inline src/libostree/ostree-sysroot-cleanup.c]: Maybe a small note at the top of this function that we're called assuming an active transaction is underway?

File: src/libostree/ostree-sysroot-cleanup.c [hunk 2]
```diff
@@ -353,34 +354,18 @@ cleanup_ref_prefix (OstreeRepo         *repo,
...
-  if (!ostree_repo_prepare_transaction (repo, NULL, cancellable, error))
-    goto out;
+    return FALSE;
 
-  g_hash_table_iter_init (&hashiter, refs);
-  while (g_hash_table_iter_next (&hashiter, &hashkey, &hashvalue))
+  GLNX_HASH_TABLE_FOREACH (refs, const char *, ref)
     {
-      const char *ref = hashkey;
       ostree_repo_transaction_set_refspec (repo, ref, NULL);
```
Comment (jlebon) [inline src/libostree/ostree-sysroot-cleanup.c]: Actually, can't we just use `ostree_repo_set_ref_immediate()` for these?

Comment (cgwalters) [reply src/libostree/ostree-sysroot-cleanup.c]: Yep, done.  Also added tests ⬇️

File: src/libostree/ostree-sysroot-cleanup.c [hunk 3]
```diff
@@ -353,34 +354,19 @@ cleanup_ref_prefix (OstreeRepo         *repo,
...
-    goto out;
-
-  ret = TRUE;
- out:
-  ostree_repo_abort_transaction (repo, cancellable, NULL);
-  return ret;
+  return TRUE;
 }
 
 static gboolean
```
Comment (jlebon) [inline src/libostree/ostree-sysroot-cleanup.c]: Am I dreaming, or did you have nice comments on top of these functions briefly explaining what they did in a previous version of this PR? I really liked them. :)

**Comment by jlebon** [comment]
Nice! @rh-atomic-bot r+ 2f6ddaa

**Comment by cgwalters** [comment]
Rebased 🏄

File: src/libostree/ostree-sysroot-cleanup.c (hunk 3 above)
Comment (cgwalters) [reply src/libostree/ostree-sysroot-cleanup.c]: Not entirely sure, but I added a comment on each function for this updated PR

**Comment by jlebon** [comment]
@rh-atomic-bot r+ b7c3d02

**Pull request closed without merging**